pkg/virt-controller/watch/migration.go

```diff
@@ -604,6 +526,92 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 	return nil
 }
 
+func (c *MigrationController) processMigrationPhase(migration, migrationCopy *virtv1.VirtualMachineInstanceMigration, pod, attachmentPod *k8sv1.Pod, podExists, attachmentPodExists bool, conditionManager *controller.VirtualMachineInstanceMigrationConditionManager, vmiConditionManager *controller.VirtualMachineInstanceConditionManager, vmi *virtv1.VirtualMachineInstance, syncError error) (*virtv1.VirtualMachineInstanceMigration, error) {
+	switch migration.Status.Phase {
+	case virtv1.MigrationPhaseUnset:
+		canMigrate, err := c.canMigrateVMI(migration, vmi)
+		if err != nil {
+			return migrationCopy, err
+		}
+
+		if canMigrate {
+			migrationCopy.Status.Phase = virtv1.MigrationPending
+		} else {
+			// can not migrate because there is an active migration already
+			// in progress for this VMI.
+			migrationCopy.Status.Phase = virtv1.MigrationFailed
+			c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "VMI is not eligible for migration because another migration job is in progress.")
+			log.Log.Object(migration).Error("Migration object ont eligible for migration because another job is in progress")
+		}
+	case virtv1.MigrationPending:
+		if podExists {
+			if controller.VMIHasHotplugVolumes(vmi) {
+				if attachmentPodExists {
+					migrationCopy.Status.Phase = virtv1.MigrationScheduling
+				}
+			} else {
+				migrationCopy.Status.Phase = virtv1.MigrationScheduling
+			}
+		} else if syncError != nil && strings.Contains(syncError.Error(), "exceeded quota") && !conditionManager.HasCondition(migration, virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota) {
+			condition := virtv1.VirtualMachineInstanceMigrationCondition{
+				Type:          virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota,
+				Status:        k8sv1.ConditionTrue,
+				LastProbeTime: v1.Now(),
+			}
+			migrationCopy.Status.Conditions = append(migrationCopy.Status.Conditions, condition)
+		}
+	case virtv1.MigrationScheduling:
+		if conditionManager.HasCondition(migrationCopy, virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota) {
+			conditionManager.RemoveCondition(migrationCopy, virtv1.VirtualMachineInstanceMigrationRejectedByResourceQuota)
+		}
+		if isPodReady(pod) {
+			if controller.VMIHasHotplugVolumes(vmi) {
+				if attachmentPodExists && isPodReady(attachmentPod) {
+					log.Log.Object(migration).Infof("Attachment pod %s for vmi %s/%s is ready", attachmentPod.Name, vmi.Namespace, vmi.Name)
+					migrationCopy.Status.Phase = virtv1.MigrationScheduled
+				}
+			} else {
+				migrationCopy.Status.Phase = virtv1.MigrationScheduled
+			}
```

Comment:
Here again, the Phase is always the same, could you please reduce to a single statement?

Reply:
I was able to reduce the code from your first comment but I am not sure exactly how to reduce this snippet of code while preserving the logging of the attachment pod being ready.